### ISSUE TITLE
Disable JIT parallelism across CIs

### ISSUE THREAD

**Issue by rsmmr** [zeek/spicy#828]
We had to add a few `-j 2` to CI builds so that the systems wouldn't run out of memory. I believe all the cases where that happened was because of Spicy's JIT compiling multiple C++ files in parallel. If that gets combined with `btest -j`, one can end up with a lot for compiler processes.

I'm thinking a better work-around than adding `-j 2` would be an environment variable that forces the JIT to compile C++ source files sequentially. We would just generally set that for CI and let the `-j` options for btest/ninja/make figure out what to spawn concurrently.

(Another option would be to teach btest an environment variable to limit concurrency to a max number of processes; but doing it on Spicy side seems more direct).

**Comment by bbannier** [comment]
Is this already covered by `HILTI_JIT_SEQUENTIAL`?

**Comment by rsmmr** [comment]
Indeed, I forgot about that and rediscovered the idea! :-)

Then task here is to change CI across the 3 repositories to use that instead of manual `-j X`.

**Comment by bbannier** [comment]
I opened zeek/spicy-plugin#20 and zeek/spicy-analyzers#25 to address this in the subprojects. For Spicy itself we already disable JIT parallelism in CI.
